src/pseudopeople/interface.py

```diff
@@ -97,6 +99,33 @@ def _generate_dataset(
     return noised_dataset
 
 
+def validate_source_compatibility(source):
+    # TODO: Clean this up w/ metadata
+    changelog = source / "CHANGELOG.rst"
+    if changelog.exists():
+        with open(changelog, "r") as file:
+            first_line = file.readline()
+        version = first_line.split("**")[1].split("-")[0].strip()
+        if version > "1.4.2":
+            raise DataSourceError(
+                "The provided input data is incompatible with this version of pseudopeople.\n"
```

Comment:
We need to change these messages to say "simulated population" instead of "input data."

And would it be possible to put the pseudopeople version in this message?

Reply:
Or "simulated population data" or even "pseudopeople simulated population data" instead of "input data". That's the terminology I used in the docstrings.

Reply:
This raises another question: Are there other error messages out there that are still using the term "input data" instead of our updated "simulated population" terminology?

Reply:
Changed "input data" - will add a todo for the version getter

Reply:
Ok, added psp version to the logs